reactor/src/main/java/com/iluwatar/reactor/app/App.java

```diff
@@ -0,0 +1,103 @@
+package com.iluwatar.reactor.app;
+
+import java.io.IOException;
+
+import com.iluwatar.reactor.framework.AbstractNioChannel;
+import com.iluwatar.reactor.framework.ChannelHandler;
+import com.iluwatar.reactor.framework.NioDatagramChannel;
+import com.iluwatar.reactor.framework.NioReactor;
+import com.iluwatar.reactor.framework.NioServerSocketChannel;
+import com.iluwatar.reactor.framework.ThreadPoolDispatcher;
+
+/**
+ * This application demonstrates Reactor pattern. The example demonstrated is a Distributed Logging Service
+ * where it listens on multiple TCP or UDP sockets for incoming log requests.
+ * 
+ * <p>
+ * <i>INTENT</i>
+ * <br/>
+ * The Reactor design pattern handles service requests that are delivered concurrently to an 
+ * application by one or more clients. The application can register specific handlers for processing
+ * which are called by reactor on specific events.
+ * 
+ * <p>
+ * <i>PROBLEM</i>
+ * <br/>
+ * Server applications in a distributed system must handle multiple clients that send them service 
+ * requests. Following forces need to be resolved:
+ * <ul>
+ * <li>Availability</li>
+ * <li>Efficiency</li>
+ * <li>Programming Simplicity</li>
+ * <li>Adaptability</li>
+ * </ul>
+ * 
+ * <p>
+ * The application utilizes single thread to listen for requests on all ports. It does not create
+ * a separate thread for each client, which provides better scalability under load (number of clients
+ * increase).
+ * 
```

Comment:
I would like to read about the participants of the pattern here. Handles, synchronous event demultiplexer, initiation dispatcher and event handlers - what class in the example implements what? I saw you have good comments on the actual classes, maybe some of that description can be put here also.